contracts/truefi2/TrueCreditAgency.sol

```diff
@@ -422,35 +422,33 @@ contract TrueCreditAgency is UpgradeableClaimable, ITrueCreditAgency {
     ) internal {
         CreditScoreBucket storage bucket = buckets[pool][bucketNumber];
 
-        poolTotalInterest[pool] = poolTotalInterest[pool].add(
-            bucket.rate.mul(1e23).mul(bucket.totalBorrowed).mul(timeNow.sub(bucket.timestamp)).div(365 days)
-        );
-
-        bucket.cumulativeInterestPerShare = bucket.cumulativeInterestPerShare.add(
-            bucket.rate.mul(ADDITIONAL_PRECISION.div(10000)).mul(timeNow.sub(bucket.timestamp)).div(365 days)
-        );
+        uint256 newInterestPerShare = _newInterestPerShare(bucket, timeNow);
+        poolTotalInterest[pool] = poolTotalInterest[pool].add(bucket.totalBorrowed.mul(newInterestPerShare));
```

Comment:
In thatI'm concerned, that in this case we do multiplication after division.

Reply:
We have 10^27 additional precision, I think we should be fine with dividing by 31556952. BTW, maybe we want to extract 1e23/(seconds in year) as const as a small optimisation?

Reply:
Hmm, that's a fair concern. But I think I'm not very worried about this numerical instability because we're still multiplying by 27 (well, actually 23) decimal points of precision inside the `_newInterestPerShare()` function. 23 decimal points of precision is pretty massive, about the same order of magnitude as counting the exact number of carbon atoms in 1 gram of diamond.

The loss of precision we get from dividing by 365 days is counteracted by the extra precision from `timeNow - bucket.timestamp` (probably a few hours?), which only takes away maybe 5 decimal points of precision.

Reply:
> BTW, maybe we want to extract 1e23/(seconds in year) as const as a small optimisation?

Not a bad idea, I'll go implement that now. If I recall correctly the Solidity compiler can optimize away simple arithmetic operations, so I might just drop the SafeMath for these constants.

Reply:
Huh, it looks like 10^23/365 days actually broke a test. I guess if we're dividing first by the seconds in a year, then integer division can truncate a lot.

Anyway, I'm splitting the division back up to regain precision. This fixed the failing test locally.